### PR TITLE
Add range match policy type for enrich

### LINKED ISSUE #59037: Support ip_range in Enrichment
Elasticsearch supports the ip_range type. 

One way to reduce a lot of traffic from investigation or focus on a specific network is by the use of allow or deny lists. 
Currently I am only able to enrich IP's individually as the enrich index converts them into a keyword.
As a subnet size increases, it is not effective to expand it and do an enrichment on a 1:1 basis.

Kindly consider for supporting ip_range as it will be helpful to reduce the size of an allow or deny list index and help quicker processing for a lot of security based use cases.

### PR DESCRIPTION
**Pull request by probakowski**
This change adds new policy types to support all range types as enrich fields. It also adds new `format` field in enrich policy definition that lets you specify date format that will be used during creation of enrich index

Closes https://github.com/elastic/elasticsearch/issues/59037

### CODE REVIEW

File: x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichProcessorFactory.java [hunk 1]
```diff
@@ -67,6 +67,11 @@ public Processor create(Map<String, Processor.Factory> processorFactories, Strin
             throw ConfigurationUtils.newConfigurationException(TYPE, tag, "max_matches", "should be between 1 and 128");
         }
 
+        if (EnrichPolicy.RANGE_MATCH_TYPES.contains(policyType)) {
+            // range match uses the same processor as simple match
```
Comment (martijnvg) [inline x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichProcessorFactory.java:71]: Maybe extend the comment here by including that a term query on a range field is translated into a range query with the values as lower and upper value. If multiple values are provided then each value is translates into a range query and that is then combined into a boolean query with should clauses, which means that on of the provided values need to fall into the range of the range field.

**Review by martijnvg — COMMENTED** [review]
In general this looks good and the change is smaller than I thought it would be. 
Can you also add docs for this new enrich policy type? 
Maybe also add a rest test to `CommonEnrichRestTestCase`?

**Review by jbaiera — COMMENTED** [review]
Had just a few comments. +1 on adding some docs, but otherwise looks pretty good.

**Comment by jakelandis** [comment]
@probakowski - can this be closed in favor of https://github.com/elastic/elasticsearch/pull/76110 ?

**Comment by mjmbischoff** [comment]
> @probakowski - can this be closed in favor of #76110 ?

The tests got reused, and I think all functionality is covered, please reopen if these assumptions are false

**Pull request closed without merging**